Add CONTRIBUTING.md to the docs

jtpavlock opened the pull request:
## Description

This does a few things:
1. Converts CONTRIBUTING.md to CONTRIBUTING.rst - this allows it to be in the docs
2. Adds CONTRIBUTING.rst to the docs
3. Adds the Testing wiki to CONTRIBUTING.rst
4. Adds a table of contents to the top of CONTRIBUTING.rst

sampsyo commented on CONTRIBUTING.rst:
```diff
@@ -0,0 +1,358 @@
+   other `testing-related
+   issues <https://github.com/beetbox/beets/labels/testing>`__.
+-  There are several ways to improve the tests in general (see
+   `Testing <https://github.com/beetbox/beets/wiki/Testing>`__) and some
```
Maybe change this to a section reference?

sampsyo commented on CONTRIBUTING.rst:
```diff
@@ -0,0 +1,358 @@
+6. Run the tests and style checker. The easiest way to run the tests is
+   to use `tox <https://tox.readthedocs.org/en/latest/>`__. For more
+   information on running tests, see our `Testing wiki
+   page <https://github.com/beetbox/beets/wiki/Testing>`__.
```
As above.

sampsyo commented on CONTRIBUTING.rst:
```diff
@@ -0,0 +1,358 @@
+-  For detailed information see `Python unittest`_
+-  **AVOID** using the ``start()`` and ``stop()`` methods of
+   ``mock.patch``, as they require manual cleanup. Use the annotation or
+   context mana
```
Looks like this got cut off somehow?

sampsyo commented on CONTRIBUTING.rst:
```diff
@@ -0,0 +1,358 @@
+call, should be skipped normally and run under our weekly `integration
+test`_ suite. These tests can be useful in detecting external changes
+that would affect ``beets``. In order to do this, simply add the
+following snippet before the applicable test case:
+
+::
```
Shorter way to do this in ReST:

```suggestion
following snippet before the applicable test case::
```

jtpavlock commented:
Thanks for the sanity check. I fixed the issues you mentioned.

> Over time, it might be nice to split this stuff up into multiple pages… there's a lot of information here, and it covers a pretty wide range from "consider writing a blog post!" to "here are the functions to use when printing a path string." We could make the "main page" a simple overview and keep the nitty-gritty details in the docs only, for example.

True, I'm hoping the table of contents I added at the top will also help with navigating around it. Either way, I think this is a good first step at getting it in the docs.

sampsyo commented:
Indeed. In any case, I think this is good to go and will hit the green button—possible reorganizations can happen "off the critical path." :smiley:
